pkg/querier/queryrange/query_range.go

```diff
@@ -52,7 +52,8 @@ var (
 type Codec interface {
 	Merger
 	// DecodeRequest decodes a Request from an http request.
-	DecodeRequest(context.Context, *http.Request) (Request, error)
+	// The last parameter specifies a list of headers on the http request that should be included in the decoded request
```

Comment:
nit: name the parameters, then refer to the name rather than "the last parameter".
Better still, use a descriptive name, then the comment can be minimal.